### PR TITLE
[FLINK-35116] Bump operator sdk version to 4.8.3

### PR DESCRIPTION
**Pull request by mbalassi**
Also bumps fabric8 client 6.11.0 to harmonize it with JOSDK.

The fabric8 client bump brings 2 notable changes to the crd-generator:
1. The fields are now sorted alphabetically due to [this](https://github.com/fabric8io/kubernetes-client/commit/18ff1e3a26a7b259dc997959867d87eafc080295). This results in a large diff, but is in principle welcome.
2. The `volumeClaimTemplate.spec.resources.claims` field has been removed from the `podTemplate`. For the time being I added it as an exception in the `CRDCompatibilityChecker`, but we need to understand its cause and implications more before merging.

Given that JOSDK 4.8.3 contains a critical fix for operator HA, I am open to splitting this into 2 separate PRs if needed.

### CODE REVIEW

**Comment by mbalassi** [comment]
I will take a look at the test failure later, it is a timeout in `AbstractFlinkServiceTest#testBlockingDeploymentDeletion`. I managed to reproduce it locally.

File: flink-kubernetes-operator-api/src/main/java/org/apache/flink/kubernetes/operator/api/validation/CrdCompatibilityChecker.java [hunk 1]
```diff
@@ -93,7 +93,10 @@ protected static void checkObjectCompatibility(
                     // This field was removed from Kubernetes ObjectMeta v1 in 1.25 as it was unused
                     // for a long time. If set for any reason (very unlikely as it does nothing),
                     // the property will be dropped / ignored by the api server.
-                    if (!fieldPath.endsWith(".metadata.clusterName")) {
+                    if (!fieldPath.endsWith(".metadata.clusterName")
+                            // This field was removed from the pod templates when upgrading
+                            // from crd-generator 6.8 to 6.11, needs further assessment
+                            && !fieldPath.contains(".volumeClaimTemplate.spec.resources.claims")) {
```
Comment (csviri) [inline flink-kubernetes-operator-api/src/main/java/org/apache/flink/kubernetes/operator/api/validation/CrdCompatibilityChecker.java:99]: This field was removed from Kubernetes API: https://github.com/kubernetes/api/blob/master/core/v1/types.go#L2620-L2625

**Comment by gyfora** [comment]
Together with the version bump, this workaround can also be removed I believe: https://github.com/apache/flink-kubernetes-operator/commit/726e484c6a9b4121563829bc094b3eebeb8ddcf3 (cc @csviri )

**Review by csviri — APPROVED** [review]
LGTM

**Pull request merged**